pkgs/applications/graphics/inkscape/default.nix

```diff
@@ -1,76 +1,91 @@
-    })
-    (fetchpatch {
-      name = "inkscape-poppler_0_82_compat.patch";
-      url = "https://gitlab.com/inkscape/inkscape/commit/835b6bb62be565efab986d5a3f30a672ad56c7eb.patch";
-      sha256 = "02c6sxi2w52b885vr3pgani6kvxp9gdqqk2jgiykkdzv70hhrnm7";
-    })
-    (fetchpatch {
-      name = "inkscape-poppler_0_83_compat.patch";
-      url = "https://gitlab.com/inkscape/inkscape/commit/b5360a807b12d4e8318475ffd0464b84882788b5.patch";
-      sha256 = "1p44rr2q2i3zkd1y1j7xgdcbgx8yvlq6hq92im8s0bkjby6p5cpz";
-    })
-  ];
-
   # Inkscape hits the ARGMAX when linking on macOS. It appears to be
   # CMake’s ARGMAX check doesn’t offer enough padding for NIX_LDFLAGS.
   # Setting strictDeps it avoids duplicating some dependencies so it
   # will leave us under ARGMAX.
   strictDeps = true;
 
-  unpackPhase = ''
-    cp $src ${name}.tar.bz2
-    tar xvjf ${name}.tar.bz2 > /dev/null
-    cd ${name}
-  '';
-
   postPatch = ''
     patchShebangs share/extensions
-    patchShebangs fix-roff-punct
+    patchShebangs share/templates
+    patchShebangs man/fix-roff-punct
 
-    # Python is used at run-time to execute scripts, e.g., those from
-    # the "Effects" menu.
+    # Python is used at run-time to execute scripts,
+    # e.g., those from the "Effects" menu.
     substituteInPlace src/extension/implementation/script.cpp \
-      --replace '"python-interpreter", "python"' '"python-interpreter", "${python2Env}/bin/python"'
+      --replace '"python3"' '"${python3Env}/bin/python"'
   '';
 
-  nativeBuildInputs = [ pkgconfig cmake makeWrapper python2Env wrapGAppsHook ]
+  nativeBuildInputs = [ pkgconfig cmake ninja python3Env wrapGAppsHook ]
     ++ (with perlPackages; [ perl XMLParser ]);
   buildInputs = [
     libXft libpng zlib popt boehmgc
-    libxml2 libxslt glib gtkmm2 glibmm libsigcxx lcms boost gettext
+    libxml2 libxslt glib gtkmm3 glibmm libsigcxx lcms boost gettext
     gsl poppler imagemagick libwpg librevenge
     libvisio libcdr libexif potrace
-
+    double-conversion
+    gdl
+    libsoup
+    pcre             # libpcre required by glib-2.0
+    libpthreadstubs  # pthread-stubs required by xcb
+    libXdmcp         # xdmcp required by xcb
+    utillinuxMinimal # mount required by gio-2.0
+    libselinux       # libselinux required by gio-2.0
+    libsepol         # libsepol required by libselinx
+    sqlite           # sqlite3 required by libsoup-2.4
+    libpsl           # libpsl required by libsoup-2.4
+    brotli           # libbrotlidec required by libsoup-2.4
+    icu              # icu-i18n required by libvisio-0.1, libcdr-0.1
+    libwpd           # libwpd-0.10 required by libwpg-0.3
+    libxkbcommon     # xkbcommon required by gdk-3.0
+    epoxy            # epoxy required by gdk-3.0
+    dbus             # dbus-1 required by atk-bridge-2.0
+    at-spi2-core     # atspi-2 required by atk-bridge-2.0
+    at-spi2-atk      # atspi-2 required by atk-bridge-2.0
+    libXtst          # xtst required by atspi-2
```

Comment:
if these are in `Requires` for pkg-config, it will need them. I'm not sure all of these are propagatedBuildInputs in their respective packages, so that may by why you had to add them for pkgconfig to work.

Reply:
My thoughts as well. That's why I prepared the full list, because I think these really belong into `propagatedBuildInputs` of the respective packages.

Reply:
I think those are just `Requires.private` and CMake complains about them but the build still continues.